src/components/rubric-levels-mobile.js

```diff
@@ -0,0 +1,1038 @@
+		return getComputedStyle(this).direction === 'rtl'
+			? this.localize('selectNextLevel')
+			: this.localize('selectPreviousLevel');
+	}
+
+	_getLevelClassName(index, selected, hovered, focused, moving) {
+		return classMap({
+			level: true,
+			selected: this._isSelected(index, selected),
+			hovered: this._isHovered(index, hovered),
+			focused: this._isFocused(index, focused) && (this.readOnly || selected === -1),
+			dragging: this._currentDragContext?.elementHasBeenDragged,
+			moving: this._currentDragContext?.elementHasBeenDragged || moving
+		});
+	}
+
+	_getLevelDescription(cell) {
+		const cellEntity = cell && cell.getSubEntityByClass && cell.getSubEntityByClass(Classes.rubrics.description);
+		return cellEntity && cellEntity.properties.text || this.localize('noDescription');
+	}
+
+	_getLevelId(index) {
+		return `level${index}`;
+	}
+
+	_getLevelIndex(index, levelCount) {
+		return getComputedStyle(this).dir === 'rtl'
+			? levelCount - index
+			: index + 1;
+	}
+
+	_getLevelName(level) {
+		return level && level.properties && level.properties.name;
+	}
+
+	_getLevelText(index, criterionCells, levelEntities) {
+		const cell = criterionCells && criterionCells[index];
+		const level = levelEntities && levelEntities[index];
+		const levelName = this._getLevelName(level);
+		const levelDesc = this._getLevelDescription(cell);
+		const score = this._getScore(level, cell);
+		const scoreOutOf = this.outOf;
+		const scoreLabel = this._getScoreLabel(score, scoreOutOf);
+		let levelText = '';
+		if (levelName) {
+			levelText += `${levelName}`;
+		}
+		if (scoreLabel) {
+			levelText += `, ${scoreLabel}`;
+		}
+		if (levelDesc) {
+			levelText += `: ${levelDesc}`;
+		}
+		return levelText;
+	}
+
+	_getLevelWidth() {
+		const firstLevel = this.shadowRoot?.querySelector('.level');
+		if (!firstLevel) {
+			return 0;
+		}
+		let levelWidth = firstLevel.getBoundingClientRect().width;
+		if (firstLevel.classList.contains('selected')) {
+			levelWidth--;
+		}
+		return levelWidth;
+	}
+
+	_getRadioAt(index) {
+		const radios = this._getRadios();
+		return radios && radios[index];
+	}
+
+	_getRadios() {
+		return this.shadowRoot.querySelectorAll('[role=radio]');
+	}
+
+	_getRadioSibling(radio, delta) {
+		const radios = this._getRadios();
+		const index = Array.prototype.indexOf.call(radios, radio);
+		return radios[index + delta];
+	}
+
+	_getRightIteratorClass(selected, focused, levels) {
+		return classMap({
+			'level-iterator-container': true,
+			disabled: !levels || !this._isRightIteratorActive(selected, focused, levels)
+		});
+	}
+
+	_getRightIteratorText() {
+		return getComputedStyle(this).direction === 'rtl'
+			? this.localize('selectPreviousLevel')
+			: this.localize('selectNextLevel');
+	}
+
+	_getScore(level, cell) {
+		const levelScore = level && level.properties && level.properties.points;
+		if (!isNaN(levelScore)) {
+			return levelScore;
+		}
+		const cellScore = cell && cell.properties && cell.properties.points;
```

Comment:
```suggestion
		const levelScore = level?.properties?.points;
		if (!isNaN(levelScore)) {
			return levelScore;
		}
		const cellScore = cell?.properties?.points;
```
These should be available from the entities as well. Not sure if we can give this function a better name or (in the future) write a more comprehensive abstraction around HM levels and cells that isn't so tightly linked to this particular webcomponent.

Reply:
Just accessed these properties from the HM entity definitions instead. I kept `level?.properties?.points` though since default level entity behavior is to return `0` if points value doesn't exist... can't remember why I did that though when I wrote the entity `get points()` accessor. 🤔   Doesn't look like we want that in this scenario.